### PR TITLE
Replace GSON by Jackson

### PR DESCRIPTION
**Pull request by infeo**

### CODE REVIEW

File: src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java [hunk 1]
```diff
@@ -184,15 +177,18 @@ private void loadKeychainEntriesIfNeeded() throws KeychainAccessException {
...
+		};
 		try (InputStream in = Files.newInputStream(keychainPath, StandardOpenOption.READ); //
 			 Reader reader = new InputStreamReader(in, UTF_8)) {
-			return Optional.ofNullable(GSON.fromJson(reader, type));
-		} catch (NoSuchFileException | JsonParseException e) {
+			return Optional.ofNullable(JSON_MAPPER.readValue(reader, type));
+		} catch (NoSuchFileException | JacksonException e) {
+			if (e instanceof JacksonException) {
+				LOG.warn("Unable to parse keychain file, overwriting existing one.");
+			}
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java:188]: Use different catch clauses instead

File: src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java [hunk 2]
```diff
@@ -184,15 +177,18 @@ private void loadKeychainEntriesIfNeeded() throws KeychainAccessException {
...
 			return Optional.empty();
-		} catch (IOException e) {
-			throw new KeychainAccessException("Could not read keychain from path " + keychainPath, e);
+		} catch (IOException ioe) {
+			throw new KeychainAccessException("Could not read keychain from path " + keychainPath, ioe);
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java:191]: Unnecessary

**Review by overheadhunter — CHANGES_REQUESTED** [review]
Looks good, just two minor code style CRs

File: src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java (hunk 2 above)
Comment (infeo) [reply src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java]: Fixed with https://github.com/cryptomator/integrations-win/pull/23/commits/0872aec65fff6b2764ba8454f0a1157ee2e69ad3

File: src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java (hunk 1 above)
Comment (infeo) [reply src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java]: Fixed with https://github.com/cryptomator/integrations-win/pull/23/commits/0872aec65fff6b2764ba8454f0a1157ee2e69ad3

**Review by overheadhunter — APPROVED** [review]

**Pull request merged**